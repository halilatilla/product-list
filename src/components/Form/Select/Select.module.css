.select {
  --cart-width: 118px;

  height: 48px;
  border: 1px solid var(--c-border);
  border-radius: 4px;
  cursor: pointer;
  padding-right: 20px;
  font-size: 17px;
  color: var(--c-text-light);
  padding-left: 8px;
  text-transform: capitalize;
}
