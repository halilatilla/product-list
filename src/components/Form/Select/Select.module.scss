.select {
  height: 48px;
  border: 1px solid var(--c-border);
  border-radius: 4px;
  cursor: pointer;
  padding-right: 20px;
  font-size: 17px;
  color: var(--c-text-light);
  padding-left: 8px;
  text-transform: capitalize;

  &:hover {
    box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.1);
  }
}
