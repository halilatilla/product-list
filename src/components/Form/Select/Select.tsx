import { FC } from 'react'
import classnames from 'classnames'

import { useAppDispatch, useAppSelector, setSortingBy } from '@src/store'

import styles from './Select.module.scss'

interface Props {
  placeholder?: string
  className?: string
  options: any[]
}

const Select: FC<Props> = ({ className, options, ...rest }) => {
  const dispatch = useAppDispatch()
  const { sortingBy } = useAppSelector((state) => state.filter)

  const handleSearchTerm = (value: string) => {
    if (!value) {
      dispatch(setSortingBy(''))
      return
    }
    dispatch(setSortingBy(value))
  }

  return (
    <select
      className={classnames(styles.select, className)}
      onChange={(e) => handleSearchTerm(e.target.value)}
      {...rest}
    >
      {options.map((option) => (
        <option key={option.value} value={option.value} selected={option.value === sortingBy}>
          {option.label}
        </option>
      ))}
    </select>
  )
}

export default Select
