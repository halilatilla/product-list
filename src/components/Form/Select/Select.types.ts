export default interface ISelect {
  placeholder?: string
  className?: string
  options: any[]
}
