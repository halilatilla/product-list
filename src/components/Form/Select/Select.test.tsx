import { render } from '@testing-library/react'

import Select from './Select'

describe('Select', () => {
  it('renders correctly', () => {
    const { container } = render(<Select />)
    expect(container).toMatchSnapshot()
  })
})
