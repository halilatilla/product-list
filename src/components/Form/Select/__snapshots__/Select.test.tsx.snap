// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
<div>
  <select
    class="select"
  >
    <option
      value="a"
    />
    <option
      value="b"
    />
  </select>
</div>
`;
